xbmc/music/windows/GUIWindowMusicBase.cpp

```diff
@@ -690,12 +690,16 @@ bool CGUIWindowMusicBase::OnContextButton(int itemNumber, CONTEXT_BUTTON button)
       Update(item->GetPath());
       int bookmark;
       m_musicdatabase.GetResumeBookmarkForAudioBook(item->GetPath(), bookmark);
-      int i=0;
-      while (i < m_vecItems->Size() && bookmark > m_vecItems->Get(i)->m_lEndOffset)
-        ++i;
-      CFileItem resItem(*m_vecItems->Get(i));
-      resItem.SetProperty("StartPercent", ((double)bookmark-resItem.m_lStartOffset)/(resItem.m_lEndOffset-resItem.m_lStartOffset)*100);
-      g_application.PlayFile(resItem, "", false);
+
+      for (int i = 0; i < m_vecItems->Size(); ++i)
+      {
+        if(bookmark <= m_vecItems->Get(i)->m_lEndOffset)
+        {
+          CFileItem resItem(*m_vecItems->Get(i));
+          resItem.SetProperty("StartPercent", ((double)bookmark-resItem.m_lStartOffset)/(resItem.m_lEndOffset-resItem.m_lStartOffset)*100);
+          g_application.PlayFile(resItem, "", false);
+        }
```

Comment:
I suggest using `std::find_if` here

btw, the missing `break` would also be solved by that :-)

Reply:
@pkerling I like your idea and redid it with `std::find_if`. Please see below.